src/Npgsql/PostgresException.cs

```diff
@@ -293,19 +292,25 @@ public override IDictionary Data
         {
             get
             {
-                // Remarks: return Dictionary with object keys although all our keys are string keys
-                // because System.Windows.Threading.Dispatcher relies on that
-                return _data ?? (_data = (
-                    from p in typeof(PostgresException).GetProperties()
-                    let k = p.Name
-                    where p.Name != nameof(Data)
-                    where p.GetCustomAttribute<PublicAPIAttribute>() != null
-                    let v = p.GetValue(this)
-                    where v != null
-                    where k != nameof(Position) && k != nameof(InternalPosition) || (int)v != 0
-                    select new { Key = k, Value = v }
-                    ).ToDictionary(kv => (object)kv.Key, kv => kv.Value)
-                );
+                if (!_isDataInitialized)
+                {
+                    foreach (var kv in
+                        from p in typeof(PostgresException).GetProperties()
+                        let k = p.Name
+                        where p.Name != nameof(Data)
+                        where p.GetCustomAttribute<PublicAPIAttribute>() != null
+                        let v = p.GetValue(this)
+                        where v != null
+                        where k != nameof(Position) && k != nameof(InternalPosition) || (int)v != 0
+                        select new { Key = k, Value = v })
+                    {
+                        base.Data[kv.Key] = kv.Value;
+                    }
+
+                    _isDataInitialized = true;
+                }
```

Comment:
I don't like this reflection based approach because we use it only for `Data` but not for other.

Reply:
Where else do you propose that we do it? The idea is that tomorrow we may add error properties, and we don't want to need to do it in many places. Any specific objection on this?

Reply:
Because you need to add that field everywhere else (serialization, `ToString`, constructor). By the way, this makes harder to support Core RT because we need to remember where we use reflection and put these types to rd.xml.

Reply:
OK, fair enough!